snow/consensus/snowball/tree.go

```diff
@@ -17,33 +17,34 @@ var (
 	_ node      = (*binaryNode)(nil)
 )
 
-func NewTree(params Parameters, choice ids.ID) Consensus {
-	snowball := newUnarySnowball(params.BetaVirtuous)
+func NewTree(cf ConsensusFactory, params Parameters, choice ids.ID) Consensus {
+	sf := cf.NewUnary(params)
 
 	t := &Tree{
 		params: params,
+		cf:     cf,
 	}
 	t.node = &unaryNode{
 		tree:         t,
 		preference:   choice,
 		commonPrefix: ids.NumBits, // The initial state has no conflicts
-		snowball:     &snowball,
+		snow:         sf,
 	}
 
 	return t
 }
 
-// Tree implements the snowball interface by using a modified patricia tree.
+// Tree implements the snow interface by using a modified patricia tree.
 type Tree struct {
-	// node is the root that represents the first snowball instance in the tree,
-	// and contains references to all the other snowball instances in the tree.
+	// node is the root that represents the first snow instance in the tree,
```

Comment:
Q: are we alright with using snow as the word for either snowflake or snowball

Reply:
We should use the same word here that we decide to go with for `nnary`: https://github.com/ava-labs/avalanchego/pull/2717#discussion_r1486614751